hcl2template/types.build.go

```diff
@@ -254,19 +254,36 @@ func (p *Parser) decodeBuildConfig(block *hcl.Block, cfg *PackerConfig) (*BuildB
 			cfg.bucket.RegisterBuildForComponent(source.String())
 		}
 
-		// Known HCP Packer Image Datasource, whose id is the SourceImageId for some build.
-		const hcpDatasourceType string = "hcp-packer-image"
+		// Known HCP Packer Image Datasource, whose id is the SourceImageId and ChannelID for some build.
 		values := ectx.Variables[dataAccessor].AsValueMap()
 
-		dsValues, ok := values[hcpDatasourceType]
+		const hcpImageDatasourceType string = "hcp-packer-image"
+		imageDatasourceValues, ok := values[hcpImageDatasourceType]
 		if !ok {
 			return build, diags
 		}
 
-		for _, value := range dsValues.AsValueMap() {
+		const hcpIterationDatasourceType string = "hcp-packer-iteration"
```

Comment:
@nywilken I want to know your thoughts about this solution to send the channel id to HCP Packer! I implemented this POC to implement HCP Packer expecting information that makes sense to send from Packer.